### ISSUE TITLE
Error InstructionListAdapter removeInstructionsFrom

### ISSUE THREAD

**Issue by wbearand** [mapbox/mapbox-navigation-android#1128]
**Android API:** N/A
**Mapbox Navigation SDK version:** 0.16.0-SNAPSHOT

### Steps to trigger behavior

 1. start navigation
 2. app crashes

### Expected behavior
why do you torture me :smile: . that the app does not crash.
### Actual behavior
App crashes

```
07-20 09:32:26.771 2030-2030/com.............................. E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com...................................., PID: 2030
    java.lang.IllegalArgumentException
        at java.util.AbstractList.subList(AbstractList.java:736)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListPresenter.removeInstructionsFrom(InstructionListPresenter.java:154)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListPresenter.updateInstructionList(InstructionListPresenter.java:146)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListPresenter.updateBannerListWith(InstructionListPresenter.java:53)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListAdapter.updateBannerListWith(InstructionListAdapter.java:49)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.updateInstructionList(InstructionView.java:918)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.updateDataFromInstruction(InstructionView.java:841)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.access$000(InstructionView.java:79)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView$1.onChanged(InstructionView.java:190)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView$1.onChanged(InstructionView.java:186)
        at android.arch.lifecycle.LiveData.considerNotify(LiveData.java:109)
        at android.arch.lifecycle.LiveData.dispatchingValue(LiveData.java:126)
        at android.arch.lifecycle.LiveData.setValue(LiveData.java:282)
        at android.arch.lifecycle.MutableLiveData.setValue(MutableLiveData.java:33)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel$1.onProgressChange(NavigationViewModel.java:262)
        at com.mapbox.services.android.navigation.v5.navigation.NavigationEventDispatcher.onProgressChange(NavigationEventDispatcher.java:144)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorThreadListener.onNewRouteProgress(RouteProcessorThreadListener.java:34)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorHandlerCallback$1.run(RouteProcessorHandlerCallback.java:97)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:7331)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by Guardiola31337** [comment]
Hey 👋 @wbearand thanks for reaching out and report your crash.

> why do you torture me 😄

Sorry about that 😬 

The fix for OP's issue started moving in https://github.com/mapbox/mapbox-navigation-android/pull/1123 and it'll be fixed when it lands.

Thanks again!

Closing here.

**Comment by wbearand** [comment]
Ok thanks @Guardiola31337 ;

I just tested 

0.16.0-SNAPSHOT/ | Fri Jul 20 14:12:00 UTC 2018

still there fyi. though slightly different. that's new : "fromIndex(0) > toIndex(-1)" .

```
2018-07-20 17:00:22.571 19531-19531/com............................. E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.........................., PID: 19531
    java.lang.IllegalArgumentException: fromIndex(0) > toIndex(-1)
        at java.util.ArrayList.subListRangeCheck(ArrayList.java:989)
        at java.util.ArrayList.subList(ArrayList.java:979)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListPresenter.removeInstructionsFrom(InstructionListPresenter.java:154)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListPresenter.updateInstructionList(InstructionListPresenter.java:146)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListPresenter.updateBannerListWith(InstructionListPresenter.java:53)
        at com.mapbox.services.android.navigation.ui.v5.summary.list.InstructionListAdapter.updateBannerListWith(InstructionListAdapter.java:49)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.updateInstructionList(InstructionView.java:918)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.updateDataFromInstruction(InstructionView.java:841)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView.access$000(InstructionView.java:79)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView$1.onChanged(InstructionView.java:190)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView$1.onChanged(InstructionView.java:186)
        at android.arch.lifecycle.LiveData.considerNotify(LiveData.java:109)
        at android.arch.lifecycle.LiveData.dispatchingValue(LiveData.java:126)
        at android.arch.lifecycle.LiveData.setValue(LiveData.java:282)
        at android.arch.lifecycle.MutableLiveData.setValue(MutableLiveData.java:33)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel$1.onProgressChange(NavigationViewModel.java:262)
        at com.mapbox.services.android.navigation.v5.navigation.NavigationEventDispatcher.onProgressChange(NavigationEventDispatcher.java:144)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorThreadListener.onNewRouteProgress(RouteProcessorThreadListener.java:34)
        at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorHandlerCallback$1.run(RouteProcessorHandlerCallback.java:97)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```


will have to rollback to 0.15.0.

**Comment by danesfeder** [comment]
@wbearand this was happening as a result of calling `instructions.indexOf(BannerInstructions)` and the provided `BannerInstructions` _not_ being in the list.  [This check](https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/list/InstructionListPresenter.java#L145-L147) was added before the PR merged and should address your `IllegalArgumentException` - please let us know if the issue persists.

**Comment by Guardiola31337** [comment]
@wbearand we've just released [`v0.16.0`](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v0.16.0) 🎉 
Feel free to test it out using ☝️ which includes the fix for your issue.
